nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/test/java/org/apache/nifi/web/controller/ControllerSearchServiceTest.java

```diff
@@ -19,537 +19,629 @@
-        Mockito.doReturn(processorName).when(processorNode1).getName();
-
-        // assign processor node to its PG
-        Mockito.doReturn(new HashSet<ProcessorNode>() {
-            {
-                add(processorNode1);
-            }
-        }).when(containingProcessGroup).getProcessors();
-    }
-
-    /**
-     * Mocks ProcessGroup due to isAuthorized(). The final class StandardProcessGroup can't be used.
-     *
-     * @param processGroupName Desired process group name
-     * @param parent           The parent process group
-     * @param authorizedToRead Can the process group data be read?
-     * @param variableRegistry The variable registry
-     * @param versionControlInformation The version control information
-     * @return Mocked process group
-     */
-    private static ProcessGroup setupMockedProcessGroup(final String processGroupName, final ProcessGroup parent, boolean authorizedToRead, final VariableRegistry variableRegistry,
-                                                        final VersionControlInformation versionControlInformation) {
-        final String processGroupId = processGroupName + "Id";
-        final ProcessGroup processGroup = mock(ProcessGroup.class);
-
-        Mockito.doReturn(processGroupId).when(processGroup).getIdentifier();
-        Mockito.doReturn(Optional.ofNullable(null)).when(processGroup).getVersionedComponentId(); // not actually searching based on versioned component id
-        Mockito.doReturn(processGroupName).when(processGroup).getName();
-        Mockito.doReturn(parent).when(processGroup).getParent();
-        Mockito.doReturn(versionControlInformation).when(processGroup).getVersionControlInformation();
-        Mockito.doReturn(variableRegistry).when(processGroup).getVariableRegistry();
-        Mockito.doReturn(parent == null).when(processGroup).isRootGroup();
-        // override process group's access rights
-        Mockito.doReturn(authorizedToRead).when(processGroup).isAuthorized(AdditionalMatchers.or(any(Authorizer.class), isNull()), eq(RequestAction.READ),
-                AdditionalMatchers.or(any(NiFiUser.class), isNull()));
-
-        return processGroup;
-    }
-
-    /**
-     * Creates a version control information using dummy attributes.
-     *
-     * @return Dummy version control information
-     */
-    private static VersionControlInformation setupVC() {
-        final StandardVersionControlInformation.Builder builder = new StandardVersionControlInformation.Builder();
-        builder.registryId("regId").bucketId("bucId").flowId("flowId").version(1);
-
-        return builder.build();
-    }
-}
+        Mockito.verify(matcherForProcessGroup, Mockito.times(searchedProcessGroups.size())).match(Mockito.any(ProcessGroup.class), Mockito.any(SearchQuery.class));
+    }
+
+    private void thenContentOfTheFollowingGroupsAreSearched(final Collection<String> searchedProcessGroups) {
```

Comment:
The parameter could be removed and replaced with a local variable:
`final Collection<String> searchedProcessGroups = processGroups.keySet();`
With that the name could be changed to `thenContentOfGroupsAreSearched`.

However not even sure how this method proves that the groups are `searched`.

Reply:
In some cases, all the groups are searched, in those scenarios the `processGrooups.keySet()` might be used. In other cases, not all the groups are searched (because of authorization), that is why in some test method there is an explicit list of groups.

I did assume that during search, the getter for a given set (collection) component will be called. Now that I look the code second time, I found a way looks better: I remained with funnel (this is a perfectly arbitrary pick). Ever group has one funnel. In the reworked version, the test checks if the funnel for the given group (which we expect to be searched) was checked with the funnel matcher. I actually found a wrong expectation from the previous version, thanks! What is your opinion?